rqt_py_common/src/rqt_py_common/rqt_roscomm_util.py

```diff
@@ -34,76 +34,83 @@
-                # printlog("setting parameter [%s]"%param.key)
-                param_server_multi.setParam(caller_id, param.key, param.value)
-            r = param_server_multi()
-            for code, msg, _ in r:
-                if code != 1:
-                    raise RLException("Failed to set parameter: %s" % (msg))
-        except RLException:
-            raise
-        except Exception as e:
-            print("load_parameters: unable to set params (last param was " +
-                  "[%s]): %s" % (param, e))
-            raise  # re-raise as this is fatal
-        rospy.logdebug("... load_parameters complete")
+        # param_server = config.master.get()
+
+        # param = None
+        # try:
+        #     # multi-call style xmlrpc
+        #     # According to API doc, get_multi() returns
+        #     # multicall XMLRPC proxy for communicating with master,
+        #     # "xmlrpclib.MultiCall"
+        #     param_server_multi = config.master.get_multi()
+
+        #     # clear specified parameter namespaces
+        #     # 2468 unify clear params to prevent error
+        #     for param in roslaunch.launch._unify_clear_params(config.clear_params):
+        #         if param_server.hasParam(caller_id, param)[2]:
+        #             param_server_multi.deleteParam(caller_id, param)
+        #     r = param_server_multi()
+        #     for code, msg, _ in r:
+        #         if code != 1:
+        #             raise RLException("Failed to clear parameter {}: ".format(msg))
+        # except RLException:
+        #     raise
+        # except Exception as e:
+        #     RqtRoscommUtil._logger.error(
+        #         "load_parameters: unable to set params (last param was [{}]): {}".format(
+        #             param, e))
+        #     raise  # re-raise as this is fatal
+
+        # try:
+        #     # multi-call objects are not reusable
+        #     param_server_multi = config.master.get_multi()
+        #     for param in config.params.values():
+        #         # suppressing this as it causes too much spam
+        #         # printlog("setting parameter [%s]"%param.key)
+        #         param_server_multi.setParam(caller_id, param.key, param.value)
+        #     r = param_server_multi()
+        #     for code, msg, _ in r:
+        #         if code != 1:
+        #             raise RLException("Failed to set parameter: %s" % (msg))
+        # except RLException:
+        #     raise
+        # except Exception as e:
+        #     print(
+        #         "load_parameters: unable to set params (last param was [%s]): %s" % (param, e))
+        #     raise  # re-raise as this is fatal
+        # RqtRoscommUtil._logger.debug("... load_parameters complete")
+        RqtRoscommUtil._logger.error("load_parameters: not implemented in ROS2))")
+        pass
 
     @staticmethod
     def iterate_packages(subdir):
         """
+        Note: Mlautman 11/2/2018
```

Comment:
Is the high level idea of this code is to get packages that contain messages/services/actions etc? Because there are some examples of that in the rcl command line package. For example, to discover packages that contain message definitions

https://github.com/ros2/ros2cli/blob/211325816e03baaa390c65551b17dece5495c509/ros2msg/ros2msg/api/__init__.py#L22

Reply:
whoa that's cool, thanks for sharing!